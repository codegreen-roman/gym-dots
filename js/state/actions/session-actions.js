import { LOAD_SESSIONS_SUCCESS } from './types'
import { compose } from 'ramda'
import data from '../../../data/sample-sessions.json'

const { sessions } = data
const noop = () => ({})

export const loadSessionsSuccess = sessions => ({
    type: LOAD_SESSIONS_SUCCESS,
    payload: {
        sessions
    }
})

export const loadSessions = () => {
    return (dispatch) => {

        const promise = new Promise((resolve) => {
            setTimeout(() => {
                resolve(sessions)
            }, 1500)
        })

        return promise
            .then(compose(dispatch, loadSessionsSuccess))
            .catch(noop)
    }
}
