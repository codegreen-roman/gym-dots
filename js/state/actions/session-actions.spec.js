import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { loadSessions, loadSessionsSuccess } from './session-actions'

const mockStore = configureMockStore([thunk])

describe('Session Actions Creators', () => {

    describe('.loadSessionsSuccess', () => {
        expect(loadSessionsSuccess()).toMatchSnapshot()
    })

    describe('.loadSessions', () => {

        const store = mockStore({ defaults: {} })

        it('creates LOAD_SESSIONS_SUCCESS action when dispatching', (done) => {

            const promise = store.dispatch(loadSessions())

            jest.runAllTimers()

            promise.then(() => {
                expect(store.getActions()).toMatchSnapshot()
                done()
            })
        })
    })

})
