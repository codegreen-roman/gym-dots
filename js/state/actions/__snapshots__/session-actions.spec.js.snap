// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "payload": Object {
    "sessions": undefined,
  },
  "type": "LOAD_SESSIONS_SUCCESS",
}
`;

exports[`Session Actions Creators .loadSessions creates LOAD_SESSIONS_SUCCESS action when dispatching 1`] = `
Array [
  Object {
    "payload": Object {
      "sessions": Array [
        Object {
          "exercises": Array [
            Object {
              "name": "Push-ups",
              "reps": 20,
              "restTime": 90,
              "sets": 5,
              "weight": 0,
            },
            Object {
              "name": "Pull-up",
              "reps": 8,
              "restTime": 60,
              "sets": 5,
              "weight": 0,
            },
            Object {
              "name": "Australian pull-ups",
              "reps": 12,
              "restTime": 30,
              "sets": 5,
              "weight": 0,
            },
          ],
          "name": "Monday",
        },
        Object {
          "exercises": Array [
            Object {
              "name": "The bench press",
              "reps": 8,
              "restTime": 90,
              "sets": 5,
              "weight": 50,
            },
            Object {
              "name": "Dumbbell flyes",
              "reps": 8,
              "restTime": 60,
              "sets": 5,
              "weight": 12,
            },
            Object {
              "name": "Around The Worlds",
              "reps": 12,
              "restTime": 60,
              "sets": 5,
              "weight": 9,
            },
          ],
          "name": "Wednesday",
        },
        Object {
          "exercises": Array [
            Object {
              "name": "Squats",
              "reps": 8,
              "restTime": 90,
              "sets": 5,
              "weight": 70,
            },
            Object {
              "duration": 60,
              "name": "Plank",
              "reps": 0,
              "restTime": 90,
              "sets": 5,
              "weight": 0,
            },
          ],
          "name": "Friday",
        },
      ],
    },
    "type": "LOAD_SESSIONS_SUCCESS",
  },
]
`;
